### ISSUE TITLE
Add missing API v1 endpoints and docs

### ISSUE THREAD

**Issue by dmorina** [crowdresearch/collective#39]
## The problem
Some of the api resources like `Qualifications`, `Ratings`, `Payments` etc have not been refactored and documented yet and one major functionality is completely missing, that is web hooks.

### Name fixes
Some of our resources do not align with REST pluralization standards. We will fix this:
- rename the qualifications resource to `qualifications` and qualification-item to `qualification-items`
- rename the `file` resource to `files`

We will also simplify the names of certain resources where there has been confusion, or could easily be confusion:
- `worker-requester-rating` resource will be migrated to `ratings`

### Expose documentation for missing endpoints
Some endpoints do not have documentation yet, and so API users are not aware they exist. We will:
- add docs for the `charges` resource, and rename this resource to `payments`, which is a more common word for what it is intended to provide
- add docs for `qualifications`
- add docs for `ratings`

### Complete missing endpoints
Some API endpoints are not yet complete. We will complete them:
-  Complete (and refactor if necessary) create/update/list endpoints for `ratings`
-  Add a utility endpoint `qualifications/:pk/items/`

### Add web hooks
Many API uses (e.g., posting and reviewing work automatically) depend on taking action when workers complete certain actions. The most common mechanism for this kind of notification is web hooks. We will:
- Add a `/web-hooks/` resource to allow users to subscribe to events that might be useful to them.
- Initially, we will support the events `project.completed` and `assignment.submitted`. Ideally we will add others, such as  `assignment.accepted` or `assignment.skipped`, that are useful to requesters.
- We will add docs for these endpoints

---
**Use comments to share your response** or use emoji 👍 to show your support. **To officially join in**, add yourself as an assignee to the proposal. **To break consensus**, comment using [this template](https://raw.githubusercontent.com/crowdresearch/collective/master/docs/breaking_consensus). To find out more about this process, read the [how-to](https://github.com/crowdresearch/collective/blob/master/docs/CONTRIBUTING.md).

**Comment by shirishgoyal** [comment]
![image](https://user-images.githubusercontent.com/315012/34331012-e36f0ec2-e8d7-11e7-9c5e-18bc779a0b54.png)

Let this fast-track train go! No Approval needed!

**Comment by dmorina** [comment]
Your point?
It’s a case where something we agreed to and is essential is not fully complete, that’s why it falls into that category of “bugs” which doesn’t need approval, that’s all ;)

Let’s try to keep comments related to the topic yeah 😅

**Comment by shirishgoyal** [comment]
I don't think that comes under the heading of "bugs". It should be "not operational" as compared to "not complete". And the title suggests it is an update to previous version and there needs to be clear proposal on what all is being changed and rationale for it.

**Comment by markwhiting** [comment]
@dmorina Nobody has assigned themselves to this. Proposals with no assignees should be closed and can be reopened when someone is willing to take them forward. If this was a mistake, i.e. you intended to assign yourself, please fix it for the next consensus period. If this was not a mistake, feel free to close this yourself.

**Comment by dmorina** [comment]
@markwhiting yeah I added the “my proposal” so I intended to add myself as assignee, sorry forgot to do that.

**Comment by dmorina** [comment]
I've updated the original proposal to clarify what changes are being proposed.
